wallet/src/main/scala/org/bitcoins/wallet/internal/UtxoHandling.scala

```diff
@@ -119,14 +118,20 @@ private[wallet] trait UtxoHandling extends WalletLogger {
                   // We should keep the utxo as reserved so it is not used in
                   // a future transaction that it should not be in
                   txo
-                case TxoState.ConfirmedReceived | TxoState.ConfirmedSpent =>
+                case TxoState.DoesNotExist | TxoState.ConfirmedReceived |
```

Comment:
Why did `TxoState.DoesNotExist` get moved here?

Reply:
`DoesNotExist` is not currently implemented, but from my understanding it will be used if a something is double spent and another txo takes it place in the chain.

I think the block getting confirmations shouldn't change it's state from DNE, we should have proper handling of if the block has been reorged back in or something.

Reply:
The intention here is that it's useful for L2 protocols like Lightning, for instance you know what the funding outpoint should be, but it hasn't actually been broadcast on the network. Same thing can apply for HTLC success txs that spend a commitment transaction. 

That should probably be the scaladoc on `TxoState.DoesNotExist` as the current one kind of sucks.

Reply:
Well the same logic applies here as well. If it does't exist yet we shouldn't be updating it to be confirmed or not